### PR TITLE
feat: Add changelog to PR body

### PR DESCRIPTION
**Pull request by msvticket**
When regenerating the docs some functionality that previously where added got added to the docs as well.

Part of fix for jenkins-x/jx#8466

### CODE REVIEW

File: pkg/promote/promote.go [hunk 1]
```diff
@@ -180,6 +182,12 @@ func NewCmdPromote() (*cobra.Command, *Options) {
 
 // AddOptions adds command level options to `promote`
 func (o *Options) AddOptions(cmd *cobra.Command) {
+	var exists bool
```
Comment (ankitm123) [inline pkg/promote/promote.go:185]: This logic can be moved to validation function. I think we should only keep the addoptions function to add options.

File: pkg/promote/promote.go [hunk 2]
```diff
@@ -188,6 +196,8 @@ func (o *Options) AddOptions(cmd *cobra.Command) {
 	cmd.Flags().StringVarP(&o.Build, "build", "", "", "The Build number which is used to update the PipelineActivity. If not specified its defaulted from  the '$BUILD_NUMBER' environment variable")
 	cmd.Flags().StringVarP(&o.Version, "version", "v", "", "The Version to promote. If no version is specified it defaults to $VERSION which is usually populated in a pipeline. If no value can be found you will be prompted to pick the version")
 	cmd.Flags().StringVarP(&o.VersionFile, "version-file", "", "", "the file to load the version from if not specified directly or via a $VERSION environment variable. Defaults to VERSION in the current dir")
+	cmd.Flags().StringVarP(&o.AddChangelog, "add-changelog", "c", "", "a file to take a changelog from to add to the pullr equest body. Typically a file generated by jx changelog.")
+	cmd.Flags().StringVarP(&o.ChangelogSeparator, "changelog-separator", "", "-----", "the separator to use between commit message and changelog in the pull request body. Default to ----- or if set the CHANGELOG_SEPARATOR environment variable")
```
Comment (ankitm123) [inline pkg/promote/promote.go:200]: You can also set the default value to `os.Getenv("CHANGELOG_SEPARATOR")`, and in the validate function just check if o.ChangelogSeparator is empty, set it to `----`

**Review by ankitm123 — COMMENTED** [review]
minor comment.

File: pkg/promote/promote.go (hunk 1 above)
Comment (msvticket) [reply pkg/promote/promote.go]: OK

File: pkg/promote/promote.go (hunk 2 above)
Comment (msvticket) [reply pkg/promote/promote.go]: I'd say it's equivalent, but for the sake of consistency, yeah.

**Review by ankitm123 — APPROVED** [review]

**Pull request merged**